libraries/AP_GPS/AP_GPS_MAV.cpp

```diff
@@ -132,12 +132,15 @@ void AP_GPS_MAV::handle_msg(const mavlink_message_t *msg)
             }
             Vector3f vel(packet.vn/100.0f, packet.ve/100.0f, packet.vd/100.0f);
             state.velocity = vel;
+            if (packet.vd != 0) {
+                state.have_vertical_velocity = true;
```

Comment:
We might need to document this type of stuff, some drivers will never go back to false, others will. I guess it depends on whether we know if vertical velocity is lost or not; here we couldn't know if velocity is back to really being 0 or if it's lost so I think this is the best approach.